src/Shape/Box.js

```diff
@@ -0,0 +1,267 @@
+      d.median = quantile(values, 0.50);
+      d.third = quantile(values, 0.75);
+
+      d.width = this._rectWidth();
+      d.height = this._rectHeight(d);
+      
+      const mode = this._whiskerMode;
+
+      if (mode[0] === "tukey") d.top = d.first - (d.third - d.first) * 1.5;
+      else if (mode[0] === "extent") d.top = max(values);
+      else if (typeof mode[0] === "number") d.top = min([max(values), quantile(values, mode[0] / 100)]);
+
+      if (mode[1] === "tukey") d.bottom = d.third + (d.third - d.first) * 1.5;
+      else if (mode[1] === "extent") d.bottom = min(values);
+      else if (typeof mode[1] === "number") d.bottom = max([min(values), quantile(values, mode[1] / 100)]);
+
+      this._topWhiskerLength = d.top - d.third;
+      this._bottomWhiskerLength = d.first - d.bottom;
+
+      d.nested = true;
+      d.__d3plusShape__ = true;
+
+      return d;
+    });
+
+    boxes.key = d => d.key;
+    return boxes;
+
+  }
+
+  /**
+      @memberof Box
+      @desc Draws the Box.
+      @param {Function} [*callback*]
+      @chainable
+  */
+  render() {
+
+    if (this._select === void 0) {
+      this.select(select("body").append("svg")
+        .style("width", `${window.innerWidth}px`)
+        .style("height", `${window.innerHeight}px`)
+        .style("display", "block").node());
+    }
+
+    const filteredData = this._dataFilter(this._data);
+
+    // Draw box.
+    new Rect()
+      .data(filteredData)
+      .x(filteredData[0].x)
+      .y(filteredData[0].y)
+      .config(this._rectConfig)
+      .select(elem("g.d3plus-box", {
+        parent: this._select
+      }).node())
+      .render();
+
+    // Compute coordinates for median line.
+    const boxTopY = this._y() - this._rectHeight(filteredData[0]) / 2;
+    const medianPoint1X = this._x() - this._rectWidth() / 2;
+    const medianPoint1Y = boxTopY + filteredData[0].third - filteredData[0].median;
+    const medianPoint2X = medianPoint1X + this._rectWidth();
+    const medianPoint2Y = medianPoint1Y;
+
+    const medianData = [
+      {x: medianPoint1X, y: medianPoint1Y},
+      {x: medianPoint2X, y: medianPoint2Y}
+    ];
+
+    // Draw median line inside the box.
+    new Line()
+      .data(medianData)
+      .select(elem("g.d3plus-box-median", {
+        parent: this._select
+      }).node())
+      .render();
+
+    // Draw 2 lines using Whisker class.
+    // Note that this._x() and this._y() are coordinates of center of the rectangle.
+
+    // Construct coordinates for top whisker startpoint.
+    const point1X = this._x();
+    const point1Y = this._y() - this._rectHeight(filteredData[0]) / 2;
+
+    // Construct coordinates for bottom whisker startpoint.
+    const point2X = this._x();
+    const point2Y = this._y() + this._rectHeight(filteredData[0]) / 2;
+
+    const whiskerData = [
+      {x: point1X, y: point1Y, length: this._topWhiskerLength, orient: "top"},
+      {x: point2X, y: point2Y, length: this._bottomWhiskerLength, orient: "bottom"}
+    ];
```

Comment:
your code here only creates a whisker for the first box, and using methods without passing data (like `this._x()`) causes errors. please create the whisker data based on `filteredData`